repl/src/main/scala/com/cloudera/livy/repl/ReplDriver.scala

```diff
@@ -83,4 +84,32 @@ class ReplDriver(conf: SparkConf, livyConf: RSCConf)
     return session.state.toString
   }
 
+  override def createWrapper(msg: BaseProtocol.BypassJobRequest): BypassJobWrapper = {
+    kind match {
+      case Spark() => super.createWrapper(msg)
+      case PySpark() | PySpark3() => {
```

Comment:
As I stated before, I think it's better if python-specific functionality is encapsulated in PythonInterpreter. e.g.

```
interpreter match {
  case pi: PythonInterpreter => pi.createWrapper(msg)
  case _ => super.createWrapper(msg)
}
```

Same for the other methods. Optimally `Interpreter` would declare these methods and you'd call them directly from the RPC handler, but that makes it hard to test the RSC code in isolation.